### PR TITLE
examples: Add new `cms-payload` example.

### PR DESCRIPTION
**Pull request by JessRynkar**
This PR adds `cms-payload` to the examples folder. The demo provides a NextJS frontend, built with Payload, which can be deployed together in a single instance.

### Improving Documentation or adding/fixing Examples

- [x] The "examples guidelines" are followed from our contributing doc https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md
- [x] Make sure the linting passes by running `pnpm build && pnpm lint`. See https://github.com/vercel/next.js/blob/canary/contributing/repository/linting.md

### CODE REVIEW

**Comment by dvakatsiienko** [comment]
The CI pipeline fails because of a change to **package.json**.
#49997

**Comment by JessChowdhury** [comment]
> The CI pipeline fails because of a change to **package.json**. #49997

I didn't change an existing package.json, only added a new one within this self contained example.

**Review by leerob — DISMISSED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 7dde3dd6a760639b7f60ff89fae60be2d3e3d760

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 7dde3dd6a760639b7f60ff89fae60be2d3e3d760

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 7dde3dd6a760639b7f60ff89fae60be2d3e3d760

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/cms-payload/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  },
+  "devDependencies": {
+    "@swc/core": "^1.3.35",
+    "@types/express": "^4.17.9",
+    "@types/node": "^18.11.18",
+    "@types/react": "^18.0.28",
+    "copyfiles": "^2.4.1",
+    "cross-env": "^7.0.3",
+    "eslint": "8.31.0",
+    "eslint-config-next": "13.1.1",
```
Comment (balazsorban44) [inline examples/cms-payload/package.json:43]:
```suggestion
```

File: examples/cms-payload/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "description": "Demo of Payload deployed as Vercel functions within NextJS",
+  "license": "MIT",
+  "scripts": {
+    "dev": "next dev",
+    "start": "next start",
+    "lint": "next lint",
```
Comment (balazsorban44) [inline examples/cms-payload/package.json:7]:
```suggestion
```

File: examples/cms-payload/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
+{
+  "description": "Demo of Payload deployed as Vercel functions within NextJS",
```
Comment (balazsorban44) [inline examples/cms-payload/package.json:2]:
```suggestion
  "description": "Demo of Payload and Next.js",
```

File: examples/cms-payload/app/(payload)/admin/[...slug]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import PayloadAdmin from "../page";
+
+// Need to render the same component for anything within /admin
+export default PayloadAdmin;
```
Comment (balazsorban44) [inline examples/cms-payload/app/(payload)/admin/[...slug]/page.tsx:4]:
```suggestion
// Need to render the same component for anything within /admin
export { default } from "../page"
```

A handy tip.

File: examples/cms-payload/app/(site)/[slug]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import React from 'react'
```
Comment (balazsorban44) [inline examples/cms-payload/app/(site)/[slug]/page.tsx:1]: This import seems to be unused. You don't need to import `React` anymore if you don't reference it, let's drop it here, and all other places as well. :+1:

**Review by balazsorban44 — DISMISSED** [review]
Thanks!

There are linting errors, please run `pnpm prettier -w examples/cms-payload` to make sure these are addressed.

File: examples/cms-payload/nodemon.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "ext": "ts",
+  "exec": "ts-node payload/server.ts"
```
[the remark this answers is not in the document]
Comment (JessChowdhury) [reply examples/cms-payload/nodemon.json]: I removed this file as we were no longer using it.

**Review by leerob — DISMISSED** [review]

**Review by ijjk — DISMISSED** [review]

**Pull request merged**